### PR TITLE
Deleted hostname from machines.yaml template.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xunpan** [comment]
@gyliu513 Please rebase

File: cmd/clusterctl/examples/ibmcloud/machines.yaml.template [hunk 1]
```diff
@@ -13,7 +13,6 @@ items:
       value:
         apiVersion: "ibmcloudproviderconfig/v1alpha1"
         kind: "IbmcloudMachineProviderSpec"
-        hostname: <host name>
```
Comment (jichenjc) [inline cmd/clusterctl/examples/ibmcloud/machines.yaml.template]: @xunpan I asked this question and at that time seems we want to use hostname+domain as name, so why we rename this ?

Comment (xunpan) [reply cmd/clusterctl/examples/ibmcloud/machines.yaml.template]: we use resource name as the host name. So, this host name is useless. Currently, this is fine for use.

And, I did not delete the internal data structure to make it a feature on demand later.

Comment (jichenjc) [reply cmd/clusterctl/examples/ibmcloud/machines.yaml.template]: ok, that works for me , just want to make sure I catch up with your changes 
thanks for all the work :)

**Pull request merged**